### PR TITLE
build: adjust the installer build and artifact staging

### PR DESCRIPTION
**Pull request by compnerd**
Adjust the installer to allow it to be built for foreign architectures. Additionally, adjust the artifact staging to include additional content that was missed previously in order to enable code signing.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @tristanlabelle @shahmishal

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

File: utils/build.ps1 [hunk 1]
```diff
@@ -1655,9 +1661,11 @@ if (-not $SkipBuild) {
 }
 
 if (-not $SkipPackaging) {
-  Build-Installer
+  Build-Installer $HostArch
 }
 
+Build-ArtifactPresentation $HostARch
```
Comment (tristanlabelle) [inline utils/build.ps1:1667]: CAsing

File: utils/build.ps1 [hunk 2]
```diff
@@ -1546,46 +1546,52 @@ function Build-DocC() {
...
 
-  if ($Stage -and (-not $ToBatch)) {
-    Copy-File "$($HostArch.BinaryCache)\installer\Release\$($HostArch.VSName)\*.cab" "$Stage\"
-    Copy-File "$($HostArch.BinaryCache)\installer\Release\$($HostArch.VSName)\*.msi" "$Stage\"
-    Copy-File "$($HostArch.BinaryCache)\installer\Release\$($HostArch.VSName)\*.msm" "$Stage\"
-    Copy-File "$($HostArch.BinaryCache)\installer\Release\$($HostArch.VSName)\installer.exe" "$Stage\"
-    # Extract installer engine to ease code-signing on swift.org CI
-    New-Item -Type Directory -Path "$($HostArch.BinaryCache)\installer\$($HostArch.VSName)\" -ErrorAction Ignore | Out-Null
-    Invoke-Program "$BinaryCache\wix-4.0.1\tools\net6.0\any\wix.exe" -- burn detach "$($HostArch.BinaryCache)\installer\Release\$($HostArch.VSName)\installer.exe" -engine "$Stage\installer-engine.exe" -intermediateFolder "$($HostArch.BinaryCache)\installer\$($HostArch.VSName)\"
+function Build-ArtifactPresentation($Arch) {
```
Comment (tristanlabelle) [inline utils/build.ps1:1581]: It's a bit of a stretch to call this a `Build` operation. Maybe, `Stage-InstallerArtifacts`?

**Review by tristanlabelle — APPROVED** [review]

File: utils/build.ps1 (hunk 2 above)
Comment (compnerd) [reply utils/build.ps1:1581]: Fair

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Pull request merged**